spock-core/src/main/java/org/spockframework/mock/runtime/mockito/MockitoMockMakerSettings.java

```diff
@@ -40,20 +41,15 @@ public MockMakerId getMockMakerId() {
     return MockMakers.mockito.getMockMakerId();
   }
 
-  void applySettings(MockSettings mockSettings) {
-    requireNonNull(mockSettings);
-    callClosure(mockitoCode, mockSettings);
-  }
-
-  private static void callClosure(Closure<?> closure, Object delegate) {
-    Closure<?> settingsClosure = (Closure<?>) closure.clone();
-    settingsClosure.setResolveStrategy(Closure.DELEGATE_FIRST);
-    settingsClosure.setDelegate(delegate);
-    settingsClosure.call(delegate);
+  void applySettings(MockSettings mockitoSettings) {
+    requireNonNull(mockitoSettings);
+    mockitoCode.setResolveStrategy(Closure.DELEGATE_FIRST);
+    mockitoCode.setDelegate(mockitoSettings);
+    GroovyRuntimeUtil.invokeClosure(mockitoCode, mockitoSettings);
```

Comment:
I think we need to `clone` the closure here as the old code did. 
Because otherwise the `setDelegate()` can be a race, if you share MockitoMockMakerSettings between multiple concurrent test executions.

Reply:
I agree, the alternative would be to `synchronize` on the closure, but I think the `clone` is the better choice. Even groovy does it, couldn't we just use `org.codehaus.groovy.runtime.DefaultGroovyMethods#with(U, groovy.lang.Closure<T>)` ?

Reply:
Oh, yeah, that was unintentional, sorry.
Didn't mean to remove the cloning.
The `DefaultGroovyMethods#with` is probably not good though, as we would then miss the exception unwrapping we do in `GroovyRuntimeUtil#invokeClosure`.